package/MDAnalysis/coordinates/DCD.py

```diff
@@ -90,440 +86,162 @@
-
-        self.dcdfile = open(self.filename, 'rb')
-        self.n_atoms = 0
-        self.n_frames = 0
-        self.fixed = 0
-        self.periodic = False
-
-        # This reads skip_timestep and delta from header
-        self._read_dcd_header()
-
-        # Convert delta to ps
-        delta = mdaunits.convert(self.delta, self.units['time'], 'ps')
-
-        self._ts_kwargs.setdefault('dt', self.skip_timestep * delta)
-        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
-        # Read in the first timestep
-        self._read_next_timestep()
-
-    def _dcd_header(self):  # pragma: no cover
-        """Returns contents of the DCD header C structure::
-             typedef struct {
-               fio_fd fd;                 // FILE *
-               fio_size_t header_size;    // size_t == sizeof(int)
-               int natoms;
-               int nsets;
-               int setsread;
-               int istart;
-               int nsavc;
-               double delta;
-               int nfixed;
-               int *freeind;
-               float *fixedcoords;
-               int reverse;
-               int charmm;
-               int first;
-               int with_unitcell;
-             } dcdhandle;
-
-        .. deprecated:: 0.7.5
-           This function only exists for debugging purposes and might
-           be removed without notice. Do not rely on it.
-
-        """
-        # was broken (no idea why [orbeckst]), see Issue 27
-        # 'PiiiiiidiPPiiii' should be the unpack string according to the struct.
-        #    struct.unpack("LLiiiiidiPPiiii",self._dcd_C_str)
-        # seems to do the job on Mac OS X 10.6.4 ... but I have no idea why,
-        # given that the C code seems to define them as normal integers
-        desc = [
-            'file_desc', 'header_size', 'natoms', 'nsets', 'setsread', 'istart',
-            'nsavc', 'delta', 'nfixed', 'freeind_ptr', 'fixedcoords_ptr', 'reverse',
-            'charmm', 'first', 'with_unitcell']
-        return dict(zip(desc, struct.unpack("LLiiiiidiPPiiii", self._dcd_C_str)))
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        return len(self._file)
 
     def _reopen(self):
-        self.ts.frame = -1
-        self._reset_dcd_read()
+        """reopen trajectory"""
+        self.ts.frame = 0
+        self._frame = -1
+        self._file.close()
+        self._file.open(self.filename.encode('utf-8'), 'r')
+
+    def _read_frame(self, i):
+        """read frame i"""
+        self._frame = i - 1
+        self._file.seek(i)
+        return self._read_next_timestep()
 
     def _read_next_timestep(self, ts=None):
-        """Read the next frame
-
-        .. versionchanged 0.11.0::
-           Native frame read into ts._frame, ts.frame naively iterated
-        """
+        """copy next frame into timestep"""
+        if self._frame == self.n_frames - 1:
+            raise IOError('trying to go over trajectory limit')
```

Comment:
It should be consistent across readers.

Perhaps time to start drafting an API on a wiki page for bare bones readers #1194 , at least so that we can keep track?